src/shared/install.c

```diff
@@ -2278,8 +2278,10 @@ static int install_context_mark_for_removal(
                                 log_debug_errno(r, "Auxiliary unit of %s not found, removing name.", i->name);
                         else {
                                 log_debug_errno(r, "Unit %s not found, removing name.", i->name);
+                                /* In case there's no unit, we still want to remove any leftover symlink, even if
+                                 * the unit might have been removed already. */
```

Comment:
hmm, this is a comment for the ENOENT condition check, maybe move down there, and add ", hence treating ENOENT as non-fatal"

Reply:
moved